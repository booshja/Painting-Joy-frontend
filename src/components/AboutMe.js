// dependencies
import React from "react";
import styled from "styled-components";

import image from "../images/Selection_002.png";
const homepage = {
    greeting: "Hi! I'm Beth!",
    message:
        "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam pharetra, velit nec aliquam pellentesque, eros velit laoreet massa, id consequat neque dolor sit amet sapien. Donec fermentum porta mi.",
};

const StyledContainer = styled.section`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 1rem 0;
    width: 90%;
`;

const StyledGreeting = styled.h2`
    text-align: center;
    font-family: "News Cycle", sans-serif;
    font-weight: 700;
    font-size: 2rem;
    letter-spacing: 1px;
    margin-bottom: 1rem;
`;

const StyledText = styled.p`
    text-align: center;
    font-family: "News Cycle", sans-serif;
    font-size: 1.4rem;
    letter-spacing: 1px;
`;

const StyledImage = styled.img`
    height: 300px;
    width: 300px;
    box-shadow: 4px 4px 4px rgba(0, 0, 0, 0.25);
`;

const StyledTextDiv = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-bottom: 1rem;
`;

const AboutMe = () => {
    return (
        <StyledContainer>
            <StyledTextDiv>
                <StyledGreeting>{homepage.greeting}</StyledGreeting>
                <StyledText>{homepage.message}</StyledText>
            </StyledTextDiv>
            <StyledImage src={image} alt="Beth" />
        </StyledContainer>
    );
};

export default AboutMe;
