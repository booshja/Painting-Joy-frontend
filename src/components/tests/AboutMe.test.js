import { getByAltText, render } from "@testing-library/react";
import AboutMe from "../AboutMe";

/** Smoke Test */
it("renders without crashing", () => {
    render(<AboutMe />);
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(<AboutMe />);
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */

it("should render correctly", () => {
    const { getByText } = render(
        <AboutMe
            greeting="Hello there!"
            message="This is a message!"
            image="http://lorempixel.com/200/200"
        />
    );
    const greeting = getByText("Hello there!");
    const message = getByText("This is a message!");

    expect(greeting).toBeInTheDocument();
    expect(message).toBeInTheDocument();
});
