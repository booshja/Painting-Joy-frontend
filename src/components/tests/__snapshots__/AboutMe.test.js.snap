// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-bdnxRM joaZrF"
  >
    <div
      class="sc-eCApnc jIfXkt"
    >
      <h2
        class="sc-gtsrHT iXpYez"
      />
      <p
        class="sc-dlnjwi cJHVwi"
      />
    </div>
    <img
      alt="The artist themselves"
      class="sc-hKFxyN htkAwe"
    />
  </section>
</DocumentFragment>
`;
